clippy_lints/src/matches.rs

```diff
@@ -470,7 +488,24 @@ pub fn overlapping<T>(ranges: &[SpannedRange<T>]) -> Option<(&SpannedRange<T>, &
 
     impl<'a, T: Copy + Ord> Ord for Kind<'a, T> {
         fn cmp(&self, other: &Self) -> Ordering {
-            self.value().cmp(&other.value())
+            match (self.value(), other.value()) {
+                (Bound::Included(a), Bound::Included(b)) |
+                (Bound::Excluded(a), Bound::Excluded(b)) => a.cmp(&b),
+                (Bound::Unbounded, _) |
+                (_, Bound::Unbounded) => unimplemented!(),
```

Comment:
Maybe add a comment that range patterns need to be bounded? I did not know that and had to check.